### ISSUE TITLE
limit cache size?

### ISSUE THREAD

**Issue by timothydmorton** [astropy/astropy#5368]
I ran into an issue when running lots of `astroquery` queries where my cache ballooned in size to ~10s of Gb---and I didn't realize there was a cache until it was enormous, so this has caused some minor headaches with my home directory quota, etc.   I raised the issue over at [astroquery](https://github.com/astropy/astroquery/issues/759) that there should be a (perhaps user-configurable) limit on the cache size, so that old stuff gets overwritten if need be?  @keflavich suggested I make this same suggestion over here.

**Comment by pllim** [comment]
Is this somehow related to #3961?

**Comment by timothydmorton** [comment]
Yes, it could in principle be related, in the sense that re-implementing a better caching scheme could also enforce a maximum cache size.  But it seems like #3961 raises lots more issues, and it seems like just the cache size limit could be implemented on its own without necessarily fixing all those as well.

**Comment by mhvk** [comment]
Hmm, checking mine, it is not GB, but even so I'd prefer the 150 MB of way out-of-date stuff was not there. So, :+1: to this suggestion, and ideally also an expiry time... (but agreed that just setting a hard limit is a good start).

**Comment by astrofrog** [comment]
👍 from me too

**Comment by kakirastern** [comment]
Is it okay for me to work on this issue if possible?

**Comment by pllim** [comment]
Caching is not for the faint-hearted, but yes, you can give it a try.

**Comment by kakirastern** [comment]
@pllim Sounds like troubles ahead for me, but anyway, I dare to give it a go...

**Comment by aarchiba** [comment]
Let me add that the cache preserves modification and access times, if the underlying filesystem tracks those, so in about five lines of code you could delete from the cache anything that hasn't been accessed in six months. (Or six months if big and a year if small, or...)
